tests/inverted-index-test.js

```diff
@@ -0,0 +1,269 @@
+  arrayOfTermsResult = searchFixture.arrayOfTermsResult,
+  multipleIndices = searchFixture.multipleIndices,
+  multipleIndicesResult = searchFixture.multipleIndicesResult,
+  multipleIndicesMultipleTermsResult = searchFixture.multipleIndicesMultipleTermsResult,
+  variedTermsResult = searchFixture.variedTermsResult,
+  fileNamedResult = searchFixture.fileNamedResult,
+  singleBookResult = searchFixture.singleBookResult,
+  searchWithFileName = searchFixture.searchWithFileName,
+  searchWithoutFileName = searchFixture.searchWithoutFileName,
+  searchWithFileNameResult = searchFixture.searchWithFileNameResult,
+  searchWithoutFileNameResult = searchFixture.searchWithoutFileNameResult,
+  searchWithoutTerms = searchFixture.searchWithoutTerms;
+
+describe('Application Tests', () => {
+  describe('Read book data tests', () => {
+    it('ensures proper response when file has invalid json in content', (done) => {
+      request
+       .post('/api/create')
+       .attach('files', 'fixtures/invalid-json-content.json')
+       .expect({ 'invalid-json-content.json':
+         { error: 'invalid json in file content' } })
+       .end((err) => {
+         if (err) {
+           return done(err);
+         }
+         done();
+       });
+    });
+    it('gives proper response for file with bad content, though valid json', (done) => {
+      request
+        .post('/api/create')
+        .attach('files', 'fixtures/no-title-or-text-field.json')
+        .expect({ 'no-title-or-text-field.json':
+          { error: 'no title or text field in book' } })
+        .end((err) => {
+          if (err) {
+            return done(err);
+          }
+          done();
+        });
+    });
+    it('gives proper response for file with bad content, though valid json array', (done) => {
+      request
+        .post('/api/create')
+        .attach('files', 'fixtures/no-title-or-text-field-2.json')
+        .expect({ 'no-title-or-text-field-2.json':
+        { error: 'no title or text field in one or more books' } })
+        .end((err) => {
+          if (err) {
+            return done(err);
+          }
+          done();
+        });
+    });
+    it('ensures proper response when file type is invalid, e.g. image files', (done) => {
+      request
+       .post('/api/create')
+       .attach('files', 'fixtures/invalid-file-type.txt')
+       .expect({ 'invalid-file-type.txt':
+        { error: 'invalid file type' } })
+       .end((err) => {
+         if (err) {
+           return done(err);
+         }
+         done();
+       });
+    });
+    it('ensures proper response when no file is uploaded', (done) => {
+      request
+        .post('/api/create')
+        .attach()
+        .expect({ error: 'no file uploaded' })
+        .end((err) => {
+          if (err) {
+            return done(err);
+          }
+          done();
+        });
+    });
+    it('ensures proper response when file is empty', (done) => {
```

Comment:
Oh thanks. I mistook empty JSON array to mean empty file. I will write a separate test for that. Thanks.